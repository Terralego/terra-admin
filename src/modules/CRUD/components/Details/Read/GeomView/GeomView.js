import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { getJSONSchemaFromGeom } from '../../../../services/utils';

import PropertyList from '../../PropertyList';

const getProperties = geometries => {
  const properties = { ...geometries };
  Object.entries(properties).forEach(([key, value], index) => {
    properties[key] = getJSONSchemaFromGeom(value, index);
  });
  return properties;
};

const GeomView = ({ geometries }) => {
  const properties = useMemo(() => getProperties(geometries), [geometries]);

  return (
    <PropertyList
      properties={properties}
      isGeom
    />
  );
};

GeomView.propTypes = {
  geometries: PropTypes.shape({}),
};

GeomView.defaultProps = {
  geometries: {},
};

export default GeomView;
