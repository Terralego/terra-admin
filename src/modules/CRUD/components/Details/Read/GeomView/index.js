import GeomView from './GeomView';

export default GeomView;
