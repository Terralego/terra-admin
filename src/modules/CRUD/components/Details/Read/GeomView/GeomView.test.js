import React from 'react';
import renderer from 'react-test-renderer';

import GeomView from './GeomView';

jest.mock('@terralego/core/modules/Map', () => ({
  CONTROL_DRAW: 'DrawControl',
  CONTROLS_TOP_LEFT: 'top-left',
}));

jest.mock('../../PropertyList', () => props => <div {...props} />);

const props = {
  geometries: {
    main: {
      geom: { type: 'Point', coordinates: [0.49820678963087, 43.9813318364992] },
      geom_type: 0,
      identifier: '123456',
      title: 'Main geometry',
      url: '/api/endpoint/to/main',
    },
    extraLayer1: {
      geom: {
        type: 'MultiPolygon',
        coordinates: [[
          [0.497868853018598, 43.98132852670764],
          [0.497943812453336, 43.98140948979532],
          [0.497990051744801, 43.98145943234657],
          [0.497868853018598, 43.98132852670764],
        ]],
      },
      geom_type: 6,
      identifier: '567890',
      title: 'Périmètre de protection',
      url: 'api/endpoint/to/extra1',
    },
    extraLayer2: {
      geom: null,
      geom_type: 6,
      identifier: null,
      title: 'Zone tampon',
      url: 'api/endpoint/to/extra2',
    },
  },
};


it('should render correctly', () => {
  let tree;
  renderer.act(() => {
    tree = renderer.create((
      <GeomView
        {...props}
      />
    ));
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
