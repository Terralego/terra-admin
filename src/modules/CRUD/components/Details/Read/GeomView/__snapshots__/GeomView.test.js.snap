// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  isGeom={true}
  properties={
    Object {
      "extraLayer1": Object {
        "display_value": Object {
          "coordinates": Array [
            Array [
              Array [
                0.497868853018598,
                43.98132852670764,
              ],
              Array [
                0.497943812453336,
                43.98140948979532,
              ],
              Array [
                0.497990051744801,
                43.98145943234657,
              ],
              Array [
                0.497868853018598,
                43.98132852670764,
              ],
            ],
          ],
          "type": "MultiPolygon",
        },
        "method": "PATCH",
        "schema": Object {
          "properties": Object {
            "coordinates": Object {
              "items": Object {
                "items": Object {
                  "items": Object {
                    "type": "number",
                  },
                  "type": "array",
                },
                "type": "array",
              },
              "type": "array",
            },
            "type": Object {
              "title": "Type",
              "type": "string",
            },
          },
          "required": Array [
            "type",
          ],
          "title": "Périmètre de protection",
          "type": "object",
        },
        "title": "Périmètre de protection",
        "type": "object",
        "ui_schema": Object {
          "coordinates": Object {
            "items": Object {
              "items": Object {
                "ui:options": Object {
                  "addable": false,
                  "orderable": false,
                  "removable": false,
                },
              },
              "ui:options": Object {
                "addable": false,
                "orderable": false,
                "removable": false,
              },
            },
            "ui:options": Object {
              "addable": false,
              "orderable": false,
              "removable": false,
            },
          },
          "type": Object {
            "ui:widget": "hidden",
          },
          "ui:field": "geometry",
        },
        "url": "api/endpoint/to/extra1",
        "value": Object {
          "coordinates": Array [
            Array [
              Array [
                0.497868853018598,
                43.98132852670764,
              ],
              Array [
                0.497943812453336,
                43.98140948979532,
              ],
              Array [
                0.497990051744801,
                43.98145943234657,
              ],
              Array [
                0.497868853018598,
                43.98132852670764,
              ],
            ],
          ],
          "type": "MultiPolygon",
        },
      },
      "extraLayer2": Object {
        "display_value": Object {
          "coordinates": Array [],
          "type": "MultiPolygon",
        },
        "method": "POST",
        "schema": Object {
          "properties": Object {
            "coordinates": Object {
              "items": Object {
                "items": Object {
                  "items": Object {
                    "items": Object {
                      "type": "number",
                    },
                    "type": "array",
                  },
                  "type": "array",
                },
                "type": "array",
              },
              "type": "array",
            },
            "type": Object {
              "title": "Type",
              "type": "string",
            },
          },
          "required": Array [
            "type",
          ],
          "title": "Zone tampon",
          "type": "object",
        },
        "title": "Zone tampon",
        "type": "object",
        "ui_schema": Object {
          "coordinates": Object {
            "items": Object {
              "items": Object {
                "items": Object {
                  "ui:options": Object {
                    "addable": false,
                    "orderable": false,
                    "removable": false,
                  },
                },
                "ui:options": Object {
                  "addable": false,
                  "orderable": false,
                  "removable": false,
                },
              },
              "ui:options": Object {
                "addable": false,
                "orderable": false,
                "removable": false,
              },
            },
            "ui:options": Object {
              "addable": false,
              "orderable": false,
              "removable": false,
            },
          },
          "type": Object {
            "ui:widget": "hidden",
          },
          "ui:field": "geometry",
        },
        "url": "api/endpoint/to/extra2",
        "value": Object {
          "coordinates": Array [],
          "type": "MultiPolygon",
        },
      },
      "main": Object {
        "display_value": Object {
          "coordinates": Array [
            0.49820678963087,
            43.9813318364992,
          ],
          "type": "Point",
        },
        "method": "PATCH",
        "schema": Object {
          "properties": Object {
            "coordinates": Object {
              "items": Object {
                "type": "number",
              },
              "type": "array",
            },
            "type": Object {
              "title": "Type",
              "type": "string",
            },
          },
          "required": Array [
            "type",
            "coordinates",
          ],
          "title": "Main geometry",
          "type": "object",
        },
        "title": "Main geometry",
        "type": "object",
        "ui_schema": Object {
          "coordinates": Object {
            "ui:options": Object {
              "addable": false,
              "orderable": false,
              "removable": false,
            },
          },
          "type": Object {
            "ui:widget": "hidden",
          },
          "ui:field": "geometry",
        },
        "url": "/api/endpoint/to/main",
        "value": Object {
          "coordinates": Array [
            0.49820678963087,
            43.9813318364992,
          ],
          "type": "Point",
        },
      },
    }
  }
/>
`;
